Config/SVTSegmentOvFile.txt

```diff
@@ -0,0 +1,25 @@
+# Copyright(c) 2019 Intel Corporation 
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+
+#====================== Segment Override File ======================
+# If the number of frames * blockPerFrame > number of QPs, the application will loop to the beginning of the file
+# Diffrence between largest and smallest value should be less or equal than 25
+# Below values are valid for 1080p video (30x17)
```

Comment:
Hi @tszumski ,

Users can set the values for the specific resolution of input frame, but we'd better consider dynamical setting, for some scenarios such as resolution change. So how about:
    1. define an array with the values: [27, 29, 31, 33, 35, 37, 39, 41, 43, 45, 47, 49, 51];
    2. Repeat assgign each element of the array into another dynamic one, with the times = picWidth / LCU_SIZE;
    3. Loop and repeat assign the next element of the array into the dynamic one. (loopCount = picHeight / LCU_SIZE).

Maybe in this way, we don't need to load from file? Unless that users would like to set other customized values.

Reply:
The point is to allow user to set customized values for each frame and LCU separately.
I have another idea to read values in different way:
    1. One input frame data could be preceded by the number of regions
    2. Each region will consist of 6 values (X1,Y1,X2,Y2, value, mode) i.e top left and bottom right corners 

As example:
"""
2 //for frame with idx=0, set 2 separate regions
0 0 300 300 40 1 // set for region (from point 0,0 to 300,300) value 40 in mode 1 (i.e EB_QP_OV_DIRECT )
900 600 1200 900 20 2// set for region (from point 900,600 to 1200,900) value 20 in mode 2 (i.e EB_QP_OV_DELTA )
1 //for frame with idx=1 set 1 region
0 0 300 300 30 1 // set for region (from point 0,0 to 300,300) value 30 in mode 1 (i.e EB_QP_OV_DIRECT )

If we reached the end of file, loop over again


How about this idea?

Reply:
I'm unclear about the typical use case or scenario. Do the users need to prepare such file with QP specificed for each LCU or region of each frame? Or is there any way (tool or module) to be used to inference different QP values for the LCUs/regions of input frames during encoding?

My suggestion is to make your idea dynamic or adaptive for different kind (size) of input frames, and easy to be used for users. Thanks!

Reply:
You are right, user need to prepare such file.
We have project/solution that create regions by neural network and object tracking with SVT-HEVC integrated (we send segmenOv without addtional files)
The whole concept for reading from file is just to prove that this enhancement works, and as a example how to integrate/use this feature